packages/date-time-utilities/src/timeMath/timeMath.ts

```diff
@@ -36,3 +36,47 @@ export const ceilMinuteToIncrement = (date: Date, increments: number) => {
   }
   return result;
 };
+
+/**
+ * Returns a date object from the selected time.
+ * @param useHour12 - If the time picker uses 12 or 24 hour formatting
+ * @param baseDate - The baseline date to calculate the offset of the selected time
+ * @param selectedTime - A string representing the user selected time
+ * @returns A new date object offset from the baseDate using the selected time.
+ */
+export const getDateFromTimeSelection = (useHour12: boolean, baseDate: Date, selectedTime: string): Date => {
```

Comment:
It would be good to add tests for this, provided it's not too hard to do so in a way that won't cause issues when running in different time zones.

Reply:
Resolved by adding tests for `getDateFromTimeSelection`.